### PR TITLE
User model

### PR DESCRIPTION
**Pull request by ElectricZebra**
Added user model with syncAndSeed

### CODE REVIEW

File: server/db/user.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    type: db.Sequelize.UUID,
+    defaultValue: db.Sequelize.UUIDV4,
+    unique: true,
+    primaryKey: true,
+  },
+  firstName: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      notNull: true,
```
Comment (leafoflegend) [inline server/db/user.js]: `allowNull` and `validate.notNull` are the same thing. Repeat for all following instances.

File: server/db/user.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  imageURL: {
+    type: Sequelize.STRING,
+    validate: {
+      isUrl: true,
+    },
+  },
+})
+
+//syncAndSeed for testing purposes
+const userSyncAndSeed = async () => {
```
Comment (leafoflegend) [inline server/db/user.js]: Fine as a testing function. Not the right place for it.

File: server/db/user.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  await db.sync({force: false});
+  await User.create({
+    firstName: 'Prof',
+    lastName: 'Katz',
+    email: 'Prof@fullstack.com',
+    password: 'whosEric'
+  })
+}
+
+userSyncAndSeed();
```
Comment (leafoflegend) [inline server/db/user.js]: Definition of a model and the seeding of a database should not be comingled.

File: server/db/user.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+const Sequelize = require('sequelize');
+const db = require('./connection');
+
+const User = db.define('user', {
+  id: {
+    type: db.Sequelize.UUID,
```
Comment (justin-cook-developer) [inline server/db/user.js]: Does db.Sequelize work? Should it just be Sequelize.UUID?

File: server/db/user.js (hunk 1 above)
Comment (justin-cook-developer) [reply server/db/user.js]: I think notEmpty is what you might have been going for. Validate against an empty string.

Comment (ElectricZebra) [reply server/db/user.js]: Thanks for the feedback. I did mean notEmpty, I'll fix that

File: server/db/user.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    },
+  },
+  lastName: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      notNull: true,
+    },
+  },
+  email: {
```
Comment (justin-cook-developer) [inline server/db/user.js]: allowNull: false for email?

File: server/db/user.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    },
+  },
+  password: {
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      notNull: false,
+    },
+  },
+  imageURL: {
```
Comment (justin-cook-developer) [inline server/db/user.js]: Maybe add a default imageURL? Something like a stock user photo.

**Pull request merged**